OneSignalSDK/onesignal/src/main/java/com/onesignal/outcomes/OSOutcomeEventsCache.java

```diff
@@ -0,0 +1,364 @@
+package com.onesignal.outcomes;
+
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteException;
+import android.support.annotation.NonNull;
+import android.support.annotation.WorkerThread;
+
+import com.onesignal.OSLogger;
+import com.onesignal.OSSharedPreferences;
+import com.onesignal.OneSignal;
+import com.onesignal.OneSignalDb;
+import com.onesignal.influence.model.OSInfluence;
+import com.onesignal.influence.model.OSInfluenceChannel;
+import com.onesignal.influence.model.OSInfluenceType;
+import com.onesignal.outcomes.OSOutcomesDbContract.CachedUniqueOutcomeTable;
+import com.onesignal.outcomes.OSOutcomesDbContract.OutcomeEventsTable;
+import com.onesignal.outcomes.model.OSCachedUniqueOutcome;
+import com.onesignal.outcomes.model.OSOutcomeEventParams;
+import com.onesignal.outcomes.model.OSOutcomeSource;
+import com.onesignal.outcomes.model.OSOutcomeSourceBody;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+class OSOutcomeEventsCache {
+
+    private static final String PREFS_OS_UNATTRIBUTED_UNIQUE_OUTCOME_EVENTS_SENT = "PREFS_OS_UNATTRIBUTED_UNIQUE_OUTCOME_EVENTS_SENT";
+
+    private OSLogger logger;
+    private OneSignalDb dbHelper;
+    private OSSharedPreferences preferences;
+
+    OSOutcomeEventsCache(OSLogger logger, OneSignalDb dbHelper, OSSharedPreferences preferences) {
+        this.logger = logger;
+        this.dbHelper = dbHelper;
+        this.preferences = preferences;
+    }
+
+    boolean getOutcomesV2IsAvailable() {
```

Comment:
Inside of `getOutcomesV2IsAvailable` what do you mean `V2Is`?

Reply:
that we have V2 activated 🤔

Reply:
OHHH read it wrong, maybe rename?

`isOutcomesV2Available` something like this?

Reply:
much better indeed!